storagedriver/azure/blockid.go

```diff
@@ -0,0 +1,56 @@
+package azure
+
+import (
+	"encoding/base64"
+	"fmt"
+	"math/rand"
+	"sync"
+
+	azure "github.com/MSOpenTech/azure-sdk-for-go/clients/storage"
+)
+
+type blockIDGenerator struct {
+	pool map[string]bool
+	m    sync.Mutex
+}
+
+// Generate returns an unused random block id and adds the generated ID
```

Comment:
@stevvooe agreed about the birthday paradox. however `pool` is there for IDs to not to collide. we generate random number until it's one that does not exist in the pool for current blob. You're probably right about seeding the `math/rand`, I seed it with time in constructor of this type. However we'll be mostly fine since generated IDs are per blob and we keep trying to find a unique ID with the help of the `pool`.

Furthermore, yes, it's not going to solve concurrency problem, however the same goes for other storage drivers as well. We don't expect multiple writers to work on the same blob, and if that happens to be the case, last writer wins.